shared/rangeanalysis/codeql/rangeanalysis/RangeAnalysis.qll

```diff
@@ -194,18 +239,40 @@ signature module Semantic {
   /** Gets the type of an expression. */
   Type getExprType(Expr e);
 
+  /**
+   * A single static single assignment (SSA) variable.
+   */
   class SsaVariable {
+    /**
+     * Gets the expression where this SSA variable is used.
+     */
     Expr getAUse();
 
+    /**
+     * Gets the basic block where this SSA variable is defined.
+     */
     BasicBlock getBasicBlock();
   }
 
+  /**
+   * A phi node in the SSA form. A phi node is a kind of node in the SSA form
+   * that represents a merge point where multiple control flow paths converge
+   * and the value of a variable needs to be selected.
+   */
```

Comment:
This comment is not really wrong, but also not terribly useful if one doesn't already know phi nodes IMHO.
"A phi node is a kind of node in the SSA form.
It has `n`inputs: phi(n_1, ..., n_max). One input for each edge e_i that reaches the phi node in the CFG.
The phi node returns n_i if control flow reaches the phi node from edge e_i.
"
Not sure whether that is really better though :D

Reply:
I'll try and improve it either with some of what you've suggested, or perhaps an example.

Reply:
I've expanded the explanation a little, stating that the value is selected according to which control flow edge was taken.  I've also added an example.  Do you think it's better now?  I'm a little reluctant to add a formula (but I'm open to it if that really makes it clearer).